src/Factory/RestControllerFactory.php

```diff
@@ -143,7 +143,6 @@ public function createServiceWithName(ServiceLocatorInterface $controllers, $nam
             ));
         }
 
-        $controller->setEventManager($events);
```

Comment:
Thanks for the hint, using a different Identifier works:
```$events->attach('ZF\Rest\Resource', 'fetch', array($this, 'onFetch'),10);```
Whether using an existing event is "a bug" is a different discussion, I am happy that our code works again.